mitiq/calibration/settings.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import product
+from typing import Any, Callable
+
+import networkx as nx
+
+from mitiq._typing import QPROGRAM
+from mitiq.benchmarks import (
+    generate_ghz_circuit,
+    generate_mirror_circuit,
+    generate_quantum_volume_circuit,
+    generate_rb_circuits,
+)
+from mitiq.pec import execute_with_pec
+from mitiq.zne import execute_with_zne
+from mitiq.zne.inference import ExpFactory, LinearFactory, RichardsonFactory
+from mitiq.zne.scaling import (
+    fold_gates_at_random,
+    fold_global,
+    insert_id_layers,
+)
+
+
+@dataclass
+class CircuitData:
+    circuit: QPROGRAM
+    dimensions: tuple[int, int]
+    type: str
+    two_qubit_gate_count: int
+
```

Comment:
What about adding here also a Strategy dataclass?
E.g.
```
@dataclass
class Strategy:
    technique: str
    technique_params: dict[str, Any]
    mitigate_function: Callable
```

This could help to bound together technique, parameters and mitigation function into a single object.